internal/datasource/config/nginx_config_parser.go

```diff
@@ -228,7 +228,13 @@ func (ncp *NginxConfigParser) createNginxConfigContext(
 					if !ncp.ignoreLog(directive.Args[0]) {
 						accessLog := ncp.accessLog(directive.Args[0], ncp.accessLogDirectiveFormat(directive),
 							formatMap)
-						nginxConfigContext.AccessLogs = ncp.addAccessLog(accessLog, nginxConfigContext.AccessLogs)
+						if ncp.agentConfig.IsMaxAccessLogFilesConfigured() {
```

Comment:
no need for the if statement here. Viper will set a default for MaxAccessLogFiles if none is set in the agent config